src/mbgl/map/transform.cpp

```diff
@@ -243,8 +243,8 @@ void Transform::flyTo(const CameraOptions &camera, const AnimationOptions &anima
 
     // When u₀ = u₁, the optimal path doesn’t require both ascent and descent.
     bool isClose = std::abs(u1) < 0.000001;
-    // Perform a more or less instantaneous transition if the path is too short.
-    if (isClose && std::abs(w0 - w1) < 0.000001) {
+    // Perform a more or less instantaneous transition if the path is too short or zoom out factor is inf.
+    if ((isClose && std::abs(w0 - w1) < 0.000001) || isinf(r(0))) {
```

Comment:
Pardon me if miss something obvious, but for me, this code doesn't compile (it does if I use explicitely std::isinf, as is done for abs)